### PR TITLE
T19 - Factorial.

### PR DESCRIPTION
**Pull request by Priyanka15-12**

### CODE REVIEW

**Comment by MetaNaveen** [comment]
Incorrect output and coding style not maintained (i.e.,. Method name)

Kindly test your code with basic test cases before submission

image

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+         Console.Write ("Enter a positive number: ");
+         if (int.TryParse (Console.ReadLine (), out int n) && n >= 0)
+            Console.Write ($"Factorial of {n} is {Factorial (n)}.");
+         else Console.WriteLine ("Invalid character. Enter a positive integer.");
+      }
+      static int Factorial (int n) {
+         //works upto 12!
+         int fact = 1;
+         for (int i = 1; i <= n; i++)
+            fact *= i;
```
Comment (MetaNaveen) [inline Program.cs:13]: You could avoid this computation for n (0, 1 and 2) by defining certain conditions before the loop itself. 
That avoids, unnecessary loop initialization, increments and computation for these inputs.

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (Priyanka15-12) [reply Program.cs:13]: Used condition for n(0,1 and 2).

**Comment by MetaNaveen** [comment]
Working fine.

**Final Notes:**
In the method, else keyword is not required as you are using return in the if statement.

**Pull request closed without merging**